### ISSUE TITLE
Control stack guard page unprotected

### ISSUE THREAD

**Issue by VoxelSpace** [dimitri/pgloader#1173]
I'm trying to migrate from MySQL to an instance of Postgres 11 in Google Cloud.  Received this error many times trying to migrate to Postgres 12 instance.  Created a new Postgres 11 instance and **it did work once**, but only once.  Even though nothing changed, I get the same error every time.

Source DB is a small (less than 10Mb) database. Biggest table has less than 1500 rows.
```
     version
-----------------
 PostgreSQL 11.6
```
pgloader command:
`pgloader --debug 'mysql://xxxxxxx:pppppppp@mysql.website.net/mysqldb' postgresql://yyyyyyyy:pppppppp@XX.XX.XX.XX/postgresdb`

debug output:
```
pgloader version 3.6.2
compiled with SBCL 2.0.1.debian
sb-impl::*default-external-format* :UTF-8
tmpdir: #P"/tmp/pgloader/"
2020-06-30T12:57:27.010000-07:00 NOTICE Starting pgloader, log system is ready.
2020-06-30T12:57:27.024000-07:00 INFO Starting monitor
2020-06-30T12:57:27.027000-07:00 LOG pgloader version "3.6.2"
2020-06-30T12:57:27.027000-07:00 INFO     SOURCE: "mysql://xxxxxxx:pppppppp@mysql.website.net/mysqldb"
2020-06-30T12:57:27.028000-07:00 INFO SOURCE URI: #<PGLOADER.SOURCE.MYSQL:MYSQL-CONNECTION mysql://xxxxxxx@mysql.website.net:3306/mysqldb {10054DD883}>
2020-06-30T12:57:27.028000-07:00 INFO     TARGET: "postgresql://yyyyyyyy:pppppppp@XX.XX.XX.XX/mysqldb"
2020-06-30T12:57:27.028000-07:00 INFO TARGET URI: #<PGLOADER.PGSQL:PGSQL-CONNECTION pgsql://yyyyyyyy@XX.XX.XX.XX:5432/mysqldb {1005700E33}>
2020-06-30T12:57:27.028000-07:00 DEBUG LOAD DATA FROM #<PGLOADER.SOURCE.MYSQL:MYSQL-CONNECTION mysql://xxxxxxx@mysql.website.net:3306/mysqldb {10054DD883}>
2020-06-30T12:57:27.259000-07:00 DEBUG CONNECTED TO #<PGLOADER.PGSQL:PGSQL-CONNECTION pgsql://yyyyyyyy@XX.XX.XX.XX:5432/mysqldb {1005700E33}>
2020-06-30T12:57:27.260000-07:00 DEBUG SET client_encoding TO 'utf8'
2020-06-30T12:57:27.390000-07:00 DEBUG SET application_name TO 'pgloader'
2020-06-30T12:57:27.796000-07:00 LOG Migrating from #<MYSQL-CONNECTION mysql://xxxxxxx@mysql.website.net:3306/mysqldb {10054DD883}>
2020-06-30T12:57:27.798000-07:00 LOG Migrating into #<PGSQL-CONNECTION pgsql://yyyyyyyy@XX.XX.XX.XX:5432/mysqldb {1005700E33}>
2020-06-30T12:57:27.918000-07:00 DEBUG CONNECTED TO #<MYSQL-CONNECTION mysql://xxxxxxx@mysql.website.net:3306/mysqldb {10054DD883}>
2020-06-30T12:57:27.925000-07:00 SQL MySQL: sending query: -- params: db-name
--         table-type-name
--         only-tables
--         only-tables
--         including
--         filter-list-to-where-clause incuding
--         excluding
--         filter-list-to-where-clause excluding
  select c.table_name, t.table_comment,
         c.column_name, c.column_comment,
         c.data_type, c.column_type, c.column_default,
         c.is_nullable, c.extra
    from information_schema.columns c
         join information_schema.tables t using(table_schema, table_name)
   where c.table_schema = 'mysqldb' and t.table_type = 'BASE TABLE'


order by table_name, ordinal_position;
2020-06-30T12:57:28.014000-07:00 SQL MySQL: sending query: -- params: db-name
--         table-type-name
--         only-tables
--         only-tables
--         including
--         filter-list-to-where-clause incuding
--         excluding
--         filter-list-to-where-clause excluding
SELECT s.table_name, s.constraint_name, s.ft, s.cols, s.fcols,
       rc.update_rule, rc.delete_rule

FROM
 (
  SELECT tc.table_schema, tc.table_name,
         tc.constraint_name, k.referenced_table_name ft,

             group_concat(         k.column_name
                          order by k.ordinal_position) as cols,

             group_concat(         k.referenced_column_name
                          order by k.position_in_unique_constraint) as fcols

        FROM information_schema.table_constraints tc

        LEFT JOIN information_schema.key_column_usage k
               ON k.table_schema = tc.table_schema
              AND k.table_name = tc.table_name
              AND k.constraint_name = tc.constraint_name

      WHERE     tc.table_schema = 'mysqldb'
            AND k.referenced_table_schema = 'mysqldb'
            AND tc.constraint_type = 'FOREIGN KEY'



   GROUP BY tc.table_schema, tc.table_name, tc.constraint_name, ft
 ) s
             JOIN information_schema.referential_constraints rc
               ON rc.constraint_schema = s.table_schema
              AND rc.constraint_name = s.constraint_name
              AND rc.table_name = s.table_name;
2020-06-30T12:57:28.075000-07:00 SQL MySQL: sending query: -- params: db-name
--         table-type-name
--         only-tables
--         only-tables
--         including
--         filter-list-to-where-clause incuding
--         excluding
--         filter-list-to-where-clause excluding
  SELECT table_name, index_name, index_type,
         sum(non_unique),
         cast(GROUP_CONCAT(column_name order by seq_in_index) as char)
    FROM information_schema.statistics
   WHERE table_schema = 'mysqldb'


GROUP BY table_name, index_name, index_type;
2020-06-30T12:57:28.128000-07:00 INFO Processing source catalogs
2020-06-30T12:57:28.381000-07:00 NOTICE Prepare PostgreSQL database.
2020-06-30T12:57:28.622000-07:00 DEBUG CONNECTED TO #<PGLOADER.PGSQL:PGSQL-CONNECTION pgsql://yyyyyyyy@XX.XX.XX.XX:5432/postgresdb {1005700E33}>
2020-06-30T12:57:28.622000-07:00 DEBUG SET client_encoding TO 'utf8'
2020-06-30T12:57:28.752000-07:00 DEBUG SET application_name TO 'pgloader'
2020-06-30T12:57:29.150000-07:00 DEBUG BEGIN
KABOOM!
INFO: Control stack guard page unprotected
Control stack guard page temporarily disabled: proceed with caution
2020-06-30T12:01:59.703000-07:00 INFO Stopping monitor

What I am doing here?

Control stack exhausted (no more space for function call frames).
This is probably due to heavily nested or infinitely recursive function
calls, or a tail call that SBCL cannot or has not optimized away.

PROCEED WITH CAUTION.
```

  - [x] pgloader --version

```
pgloader version "3.6.1"
compiled with SBCL 1.4.15.debian
```

also tried
```
pgloader version "3.6.2"
compiled with SBCL 2.0.1.debian
```
  
  - [x] did you test a fresh compile from the source tree?
    
Yes, 3.6.2 as noted above was compiled locally.
    
  - [x] did you search for other similar issues?
Yes

  - [ ] how can I reproduce the bug?
I'm not sure.

**Comment by dimitri** [comment]
Can you please paste the Postgres version string as per the following SQL query: `SELECT version()`. I think that's where this problem comes from, and it has been fixed in the current master's branch of pgloader.

**Comment by VoxelSpace** [comment]
> Can you please paste the Postgres version string as per the following SQL query: `SELECT version()`.
```
     version
-----------------
 PostgreSQL 11.6
```
Just another update.  I installed Postgres 11.8 locally and pgloader works great.  Although I had to revert back to 3.6.1 as the cast I needed (`--cast "type date to date using zero-dates-to-null"`) did not work with 3.6.2 (but that's a different issue).

So I'm not sure if it is a connection issue to Google cloud or something in Postgres 11.6 specifically.
